src/NuxeoClient/Client.cs

```diff
@@ -566,6 +566,14 @@ internal MultipartContent BuildMultipartContent(JToken data)
 
         private void AddBlobToMultipartContent(MultipartContent content, Blob blob)
         {
+            if (blob.File == null)
+            {
+                throw new ArgumentNullException("blob.File", "input blob has no file");
+            }
+            if (!blob.File.Exists)
```

Comment:
else if?

Reply:
I guess it'd would be redundant since it won't be executed if the first one if goes through.